run_benchmarks.rb

```diff
@@ -32,21 +32,37 @@ def os
 end
 
 # Checked system - error if the command fails
-def check_call(command, verbose: false, env: {})
+def check_call(command, env: {})
   puts(command)
 
-  if verbose
-    status = system(env, command, out: $stdout, err: :out)
-  else
-    status = system(env, command)
-  end
+  status = system(env, command)
 
   unless status
     puts "Command #{command.inspect} failed in directory #{Dir.pwd}"
     raise RuntimeError.new
   end
 end
 
+# Call system, capture stderr, return that with result.
+def run_benchmark_command(command, env: {})
+  puts(command)
+
+  result = {}
+
+  IO.pipe do |reader, writer|
```

Comment:
So I did https://github.com/Shopify/yjit-bench/pull/280, and it works for `--yjit-stats`. But then I also noticed I sometimes print debugging logs to stderr (not stdout so that it doesn't influence tests, etc.). The capturing seems to make it pretty slow, and you don't get any output since the output is no longer streamed.

Can we do either of the following two options?

1. The stderr output is streamed to the console, and show the captured result at the end again.
2. Just avoid capturing stderr.

To my knowledge, there's no easy way to implement (1). You'd need to spawn a Thread to achieve it like Open3 does. Because of the complexity, I'm personally in favor of (2); simply stop capturing `result[:err]`.

WDYT?

Reply:
:+1: to avoid capturing stderr, it's enough to scroll above to find the error(s) and definitely do not want to delay those.

Reply:
Yeah, with the summary at the end making it obvious that some failed, even if you can't scrollback you can always run individual ones again to see the errors.  Always happy to make it simpler: https://github.com/Shopify/yjit-bench/pull/281